plugins/outputs/redistimeseries/redists.go

```diff
@@ -0,0 +1,82 @@
+package redistimeseries
+
+import (
+	"github.com/go-redis/redis/v7"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+var sampleConfig = `
+  ## The address of the RedisTimeSeries server.
+  addr = "127.0.0.1:6379"
+
+  ## password to login Redis
+  # password = ""
+`
+
+type RedisTimeSeries struct {
+	addr     string
+	password string
+	client   *redis.Client
+}
+
+func (i *RedisTimeSeries) Connect() error {
+
+	client := redis.NewClient(&redis.Options{
+		Addr:     i.addr,
+		Password: i.password,
+		DB:       0, // use default DB
+	})
+
+	err := client.Ping().Err()
+	i.client = client
+	return err
+}
+
+func (i *RedisTimeSeries) Close() error {
+	return i.client.Close()
+}
+
+func (i *RedisTimeSeries) Description() string {
+	return "Configuration for sending metrics to RedisTimeSeries"
+}
+
+func (i *RedisTimeSeries) SampleConfig() string {
+	return sampleConfig
+}
+func (i *RedisTimeSeries) Write(metrics []telegraf.Metric) error {
+
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	for _, m := range metrics {
+		now := m.Time().UnixNano() / 1000000000
+		//		tags := m.Tags() TODO add support for tags
+		name := m.Name()
+
+		for fieldName, value := range m.Fields() {
+			key := name + "_" + fieldName
+			err := i.client.Do("TS.ADD", key, now, value).Err()
```

Comment:
If I do understand [the documentation](https://oss.redislabs.com/redistimeseries/commands/#tsadd) of `TS.ADD` correctly the timeseries is created if it does not exist. So why do you try to do this manually below?